lib/shopify_api/graphql/bulk_fetch.ex

```diff
@@ -0,0 +1,201 @@
+defmodule ShopifyAPI.GraphQL.BulkFetch do
+  alias ShopifyAPI.AuthToken
+
+  @max_poll_count 100
+  @polling_timeout_message "BulkFetch timed out before completion"
+  @polling_query """
+  {
+    currentBulkOperation {
+      id
+      status
+      errorCode
+      createdAt
+      completedAt
+      objectCount
+      fileSize
+      url
+      partialDataUrl
+    }
+  }
+  """
+
+  @doc """
+  ## Example
+      iex> prod_id = 10
+      iex> query = \"""
+        {
+          product(id: "gid://shopify/Product/\#{prod_id}") {
+            collections(first: 1) {
+              edges {
+                node {
+                  collection_id: id
+                  }
+                }
+              }
+            metafields(first: 1) {
+              edges {
+                node {
+                  key
+                  value
+                  metafield_id: id
+                }
+              }
+            }
+          }
+        }
+      \"""
+      iex> {:ok, token} = YourShopifyApp.ShopifyAPI.Shop.get_auth_token_from_slug("slug")
+      iex> ShopifyAPI.GraphQL.BulkFetch.process!(token, query)
+      {:ok,
+        [%{"collection_id" => "gid://shopify/Collection/xxx", ...}],
+      }
+  """
+  @spec process!(AuthToken.t(), String.t(), integer()) :: {:ok, list()} | {:error, any()}
+  def process!(%AuthToken{} = token, query, polling_rate \\ 100) do
+    case fetch_jsonl(token, query, polling_rate) do
+      {:ok, jsonl} -> {:ok, parse_bulk_response!(jsonl)}
+      {:error, _} = error -> error
+    end
+  end
+
+  @doc """
+  Like process/3 but returns a Streamable collection of decoded JSON or errors.
+
+    ## Example
+      iex> prod_id = 10
+      iex> query = \"""
+        {
+          product(id: "gid://shopify/Product/\#{prod_id}") {
+            collections(first: 1) {
+              edges {
+                node {
+                  collection_id: id
+                  }
+                }
+              }
+            metafields(first: 1) {
+              edges {
+                node {
+                  key
+                  value
+                  metafield_id: id
+                }
+              }
+            }
+          }
+        }
+      \"""
+      iex> {:ok, token} = YourShopifyApp.ShopifyAPI.Shop.get_auth_token_from_slug("slug")
+      iex> func = fn {:ok, json} -> IO.puts(json) end # Warning we are not handling error conditions here.
+      iex> ShopifyAPI.GraphQL.BulkFetch.process_stream(token, query, func)
```

Comment:
This is out of date now, the process_stream/3 returns the Enumerable so this should read

```
ShopifyAPI.GraphQL.BulkFetch.process_stream(token, query) |> func()
```